debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresTaskContext.java

```diff
@@ -111,6 +111,8 @@ protected ReplicationConnection createReplicationConnection(boolean exportSnapsh
         return ReplicationConnection.builder(config.jdbcConfig())
                 .withSlot(config.slotName())
                 .withPublication(config.publicationName())
+                .withTableWhitelist(config.tableWhitelist())
```

Comment:
This needs to consider also the table blacklist as well as database blacklist/whitelist. I suppose this was always an implicit assumption on my end during the conversations, sorry for not being more clear on that. You can take a look at `RelationalSnapshotChangeEventSource.determineCapturedTables(RelationalSnapshotContext)` to see how that's handled during snapshotting using the `TableFilter` contract.